src/Arcus.Messaging.Abstractions/MessageContext.cs

```diff
@@ -35,7 +17,7 @@ public MessageContext(string messageId, IDictionary<string, object> properties)
         /// <param name="properties">The contextual properties provided on the message.</param>
         /// <exception cref="ArgumentException">Thrown when the <paramref name="messageId"/> or the <paramref name="jobId"/> is blank.</exception>
         /// <exception cref="ArgumentNullException">Thrown when the <paramref name="properties"/> is <c>null</c>.</exception>
-        public MessageContext(string messageId, string jobId, IDictionary<string, object> properties)
+        protected MessageContext(string messageId, string jobId, IDictionary<string, object> properties)
         {
             if (string.IsNullOrWhiteSpace(messageId))
```

Comment:
Since we're on this class anyway, we can also use the new `ArgumentException.Throw...` extensions to validate the incoming parameters.

Reply:
All arguments of that method are now being checked by the new extension methods instead of the oldschool way. I will also need new tasks or issues to pick up :)